tubesync/sync/models.py

```diff
@@ -482,6 +489,15 @@ def type_directory_path(self):
     def make_directory(self):
         return os.makedirs(self.directory_path, exist_ok=True)
 
+    @property
+    def get_thumbnail_url(self):
+        if self.source_type==Source.SOURCE_TYPE_YOUTUBE_PLAYLIST:
+            raise Exception('This source is a playlist so it doesn\'t have thumbnail.')
+        soup = BeautifulSoup(requests.get(self.url, cookies={'CONSENT': 'YES+1'}).text, "html.parser")
```

Comment:
This is the only concern I have with the commit. How stable is using `requests.get` to load pages from YouTube? This needs confirmation it's reliable, as well as wrapping in some exception catches.

Reply:
Oh and the HTTP status code probably needs a check.